### PR TITLE
split user setting step from cucushift-pre

### PR DESCRIPTION
**Pull request by jianzhangbjz**
To follow the Prow rules and create a basic step framework for the user setting, I split the user setting step from the `cucushift-pre`. Cover #32579 
@xingxingxia @liangxia Could you help have a review when you get a chance? Thanks!

### CODE REVIEW

**Comment by xingxingxia** [comment]
/lgtm
Thanks!

**Comment by jianzhangbjz** [comment]
Looks good, see: https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/32626/rehearse-32626-periodic-ci-openshift-openshift-tests-private-release-4.12-nightly-4.12-upgrade-from-stable-4.11-aws-ipi-byo-route53-p2/1574637662952755200/artifacts/aws-ipi-byo-route53-p2/user-setting-htpasswd/build-log.txt 
/assign @danilo-gemoli

File: ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    if [ -f "${SHARED_DIR}/runtime_env" ] ; then
+    source "${SHARED_DIR}/runtime_env"
+    fi
+    runtime_env=${SHARED_DIR}/runtime_env
+    users=${users::-1}
+
+
+    cat <<EOF >>"${runtime_env}"
+export USERS=${users}
+EOF
```
Comment (xingxingxia) [inline ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: @jianzhangbjz , found above ci/rehearse/periodic-ci-openshift-verification-tests-master-nightly-4.11-upgrade-from-stable-4.10-from-stable-4.9-aws-ipi [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/32626/rehearse-32626-periodic-ci-openshift-verification-tests-master-nightly-4.11-upgrade-from-stable-4.10-from-stable-4.9-aws-ipi/1574606738970120192/build-log.txt) show `USERS: unbound variable`:
```
'/tmp/kubeconfig-3084384126' -> '/tmp/kubeconfig'
/bin/bash: line 19: USERS: unbound variable
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-09-27T04:36:34Z"}
error: failed to execute wrapped command: exit status 1
```

Comment (xingxingxia) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: In this new step, seems you need explicitly run `export USERS=${users}` like originally `cucushift-pre` does.

Comment (xingxingxia) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: If you will have new update, could you indent line 67 BTW?

Comment (jianzhangbjz) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: Hi @xingxingxia , as you can see from https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/32626/rehearse-32626-periodic-ci-openshift-verification-tests-master-nightly-4.11-upgrade-from-stable-4.10-from-stable-4.9-aws-ipi/1574606738970120192/build-log.txt, this step `user-setting-htpasswd` not be run. So, we should update this `chain` to call this new step.

File: ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+ ref:
+  as: user-setting-htpasswd
```
Comment (liangxia) [inline ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-ref.yaml:2]: For the folder name `user-setting`, how about use `idp` or `identity-providers` ?

Comment (jianzhangbjz) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-ref.yaml:2]: @xingxingxia What do you think?

Comment (xingxingxia) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-ref.yaml:2]: Good suggestion.

File: ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh (hunk 1 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: The below error occurred when running https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/prepare/cucushift-upgrade-prepare-commands.sh#L19
```yaml
[36mINFO[0m[2022-09-27T04:36:42Z] Logs for container test in pod aws-ipi-cucushift-upgrade-prepare: 
[36mINFO[0m[2022-09-27T04:36:42Z] '/tmp/kubeconfig-3084384126' -> '/tmp/kubeconfig'
/bin/bash: line 19: USERS: unbound variable
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-09-27T04:36:34Z"}
error: failed to execute wrapped command: exit status 1 
```

Comment (jianzhangbjz) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: The root cause is this `user-setting-htpasswd` weren't be called.

Comment (jianzhangbjz) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: That's weird! Not sure why this new step `user-setting-htpasswd` wasn't called. As you can see below, this `cucushift-upgrade-prepare` ran after the `cucushift-pre`, but it should be run after `user-setting-htpasswd`. I guess it's a flake. @danilo-gemoli Any suggestions? Thanks!
image

image

Comment (jianzhangbjz) [reply ci-operator/step-registry/user-setting/htpasswd/user-setting-htpasswd-commands.sh:75]: Another concern, is why not it failed at line 18?
image

File: ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml [hunk 3]
```diff
@@ -2,6 +2,7 @@ chain:
   as: cucushift-upgrade
   steps:
   - ref: cucushift-pre
+  - ref: user-setting-htpasswd
```
Comment (xingxingxia) [inline ci-operator/step-registry/cucushift/upgrade/cucushift-upgrade-chain.yaml:5]: > That's weird! Not sure why this new step user-setting-htpasswd wasn't called

@jianzhangbjz for your this question in previous comment, seems it is due to line 4 and 5 should swap order?

**Comment by xingxingxia** [comment]
/lgtm
LGTM, but let's wait for other guys' inputs on https://github.com/openshift/release/pull/32626#discussion_r983008053 issue if still occurring in new tests' results.

**Comment by jianzhangbjz** [comment]
Hi @danilo-gemoli, could you help approve it? Thanks!
/assign @danilo-gemoli

**Pull request merged**